common/compatibility/checker.php

```diff
@@ -0,0 +1,117 @@
+ * and the client's browser are compatible with OpenEMR. Please note
+ * that user agent sniffing is used to determine the browser. This is
+ * generally a bad practice, but we must ensure users are using only
+ * the officially supported/tested browsers.
+ *
+ * Note that this will only be used within setup.php, sql_upgrade.php,
+ * sql_patch.php, acl_upgrade.php, admin.php, and interface/login/login.php.
+ *
+ * Copyright (C) 2017 Matthew Vita <matthewvita48@gmail.com>
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>.
+ *
+ * @package OpenEMR
+ * @author  Matthew Vita <matthewvita48@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+// No namespace here because this class is mostly required outside of the
+// OpenEMR context.
+
+class Checker {
+    /**
+     * This is needed per the modernization project.
+     */
+    private $minimumPhpVersion = "5.4.0";
+
+    /**
+     * List of officially supported/tested browsers. Versions
+     * aren't specified because these browsers tend to self
+     * update/encourage end users to stay current.
+     */
+    private $supportedBrowsers = array("Firefox",
+                                       "Chrome");
+
+    /**
+     * Default constructor.
+     */
+    public function __construct() {}
+
+    /**
+     * Determines if a message should be sent to the user
+     * about their PHP / browser compatibility.
+     *
+     * @return empty string or string with compatability warnings.
+     */
+    public function getCompatibility() {
+        function xlDelegate($value) {
+            if (function_exists("xl")) {
+                return xl($value);
+            }
+
+            return $value;
+        }
+
+        $phpCheck = $this->isPhpSupported();
+        $browserCheck = $this->isBrowserSupported();
+        $response = "";
+
+        if (!$phpCheck) {
+            $response .= xlDelegate("PHP version needs to be at least") . " " . $this->minimumPhpVersion . ".";
+        }
+
+        if (!$browserCheck) {
+            if (!empty($response)) {
+                $response .= "\\n\\n";
+            }
+
+            $response .= xlDelegate("Your browser needs to be one of the following") . ": ";
+            $response .= "[" . implode(", ", $this->supportedBrowsers) . "] ";
+            $response .= xlDelegate("The OpenEMR development team exclusively tests " .
```

Comment:
Shouldn't break lines. Each line goes as a separate constant for the translators, so best to keep them as one line (even if it is a paragraph of text).
``````
xlDelegate("The OpenEMR development team exclusively tests with modern versions of these supported browsers")...